File: ios/Sources/ZappMessaging/Models/ZMPushTopicSnapshot.swift
import Foundation

public struct ZMPushTopicSnapshot: Sendable {
    public let version: Int
    public let hydrated: Bool
    public let supportsGroups: Bool
    public let conversations: [ZMPushConversationTopics]
}

public struct ZMPushConversationTopics: Sendable {
    public let conversationId: String
    public let lifecycle: String
    public let inboundTopics: [ZMPushTopic]
}

public struct ZMPushTopic: Sendable {
    public let topic: String
    public let writerPublicKey: String
}

File: android/src/main/java/xyz/justzappit/zappmessaging/models/ZMPushTopicSnapshot.kt
package xyz.justzappit.zappmessaging.models

data class ZMPushTopicSnapshot(
    val version: Int,
    val hydrated: Boolean,
    val supportsGroups: Boolean,
    val conversations: List<ZMPushConversationTopics>,
)

data class ZMPushConversationTopics(
    val conversationId: String,
    val lifecycle: String,
    val inboundTopics: List<ZMPushTopic>,
)

data class ZMPushTopic(
    val topic: String,
    val writerPublicKey: String,
)
